File: 06-06-2025/QuizupAPI/Models/Teacher.cs

using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using QuizupAPI.Interfaces;

namespace QuizupAPI.Models
{
    public class Teacher : ISoftDeletable
    {
        public long Id { get; set; }
        [Required]
        public string Email { get; set; } = string.Empty;
        [Required]
        public string FirstName { get; set; } = string.Empty;
        [Required]
        public string LastName { get; set; } = string.Empty;
        
        public ICollection<TeacherSubject>? TeacherSubjects { get; set; }
        public ICollection<TeacherClass>? TeacherClasses { get; set; }

        public bool IsDeleted { get; set; } = false;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }
        public User? User { get; set; }
        public ICollection<Quiz>? Quizzes { get; set; }
    }
}

File: Quizup-Angular-Application/src/app/features/teacher/models/teacher.ts
import { TeacherClassModel } from '../../../shared/models/teacher-class.model';
import { TeacherSubjectModel } from '../../../shared/models/teacher-subject.model';
import { ClassModel } from '../../class/models/class';
import { QuizModel } from '../../quiz/models/quiz.model';

export class TeacherModel {
  constructor(
    public id: string = '',
    public email: string = '',
    public firstName: string = '',
    public lastName: string = '',
    public teacherSubjects?: TeacherSubjectModel[],
    public teacherClasses?: TeacherClassModel[],
    public classes?: ClassModel[],
    public createdAt: Date = new Date(),
    public quizzes?: QuizModel[]
  ) {}
}
